### PR TITLE
Bugfix - Added unit tests for common input patterns

### PR DESCRIPTION
**Pull request by oyo**
Note: no entry in change log because this does not change functionality.

currently testing:

- BPNs
- mail addresses
- domain names
- URLs
- UUIDs

### CODE REVIEW

File: cx-portal/src/types/Patterns.ts [hunk 1]
```diff
@@ -21,7 +21,9 @@
 export const Patterns = {
   BPN: /^(BPNL|CAX)[0-9A-Z]{12}$/i,
```
Comment (jjeroch) [inline cx-portal/src/types/Patterns.ts:22]: we can kick out the CAX that was only a test value

File: cx-portal/src/types/Patterns.test.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      '   donald.duck@bmw.de',
+      'julia.jeroch@bmw',
+      '.eins.zwei.@drei.my',
+      'a@b.c',
+      'not,a@valid@address.com',
+      '@mickey.mouse',
+      '&$%/&()/@&/().&/',
+      '',
+    ],
+    DOMAIN: [
```
Comment (jjeroch) [inline cx-portal/src/types/Patterns.test.ts:60]: where do we use the domain? Does it make sense to combine it with our prefix "https"

File: cx-portal/src/types/Patterns.ts (hunk 1 above)
Comment (oyo) [reply cx-portal/src/types/Patterns.ts]: done

Comment (jjeroch) [reply cx-portal/src/types/Patterns.ts]: thx

File: cx-portal/src/types/Patterns.test.ts (hunk 2 above)
Comment (jjeroch) [reply cx-portal/src/types/Patterns.test.ts]: :) nice

File: cx-portal/src/types/Patterns.test.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      '&$%/.&()/@&./().&/',
+      '',
+    ],
+  },
+  URL: {
+    valid: [
+      'https://www.bmw.com',
+      'http://WWW.GOOGLE.COM/',
+      'https://www.bmw.com/index.html',
+      'http://hostname.domain:12345',
```
Comment (jjeroch) [inline cx-portal/src/types/Patterns.test.ts:80]: @oyo last question before merging - why should we allow http. I would say we only allow https

Comment (oyo) [reply cx-portal/src/types/Patterns.test.ts]: @jjeroch done - restricted to https now and also added tests to validate UUIDs

**Review by jjeroch — APPROVED** [review]

**Pull request merged**